exporter/latency.go

```diff
@@ -30,9 +35,77 @@ func (e *Exporter) extractLatencyMetrics(ch chan<- prometheus.Metric, c redis.Co
 			var spikeLast, spikeDuration, max int64
 			if _, err := redis.Scan(latencyResult, &eventName, &spikeLast, &spikeDuration, &max); err == nil {
 				spikeDurationSeconds := float64(spikeDuration) / 1e3
-				e.registerConstMetricGauge(ch, "latency_spike_last", float64(spikeLast), eventName)
-				e.registerConstMetricGauge(ch, "latency_spike_duration_seconds", spikeDurationSeconds, eventName)
+				exporter.registerConstMetricGauge(outChan, "latency_spike_last", float64(spikeLast), eventName)
+				exporter.registerConstMetricGauge(outChan, "latency_spike_duration_seconds", spikeDurationSeconds, eventName)
 			}
 		}
 	}
 }
+
+func extractLatencyHistogramMetrics(exporter *Exporter, outChan chan<- prometheus.Metric, redisConn redis.Conn) {
+	reply, err := redis.Values(doRedisCmd(redisConn, "LATENCY", "HISTOGRAM"))
+	if err != nil {
+		logHistogramErrOnce.Do(func() {
+			log.Errorf("WARNING, LOGGED ONCE ONLY: cmd LATENCY HISTOGRAM, err: %s", err)
+		})
+		log.Debugf("cmd LATENCY HISTOGRAM, err: %s", err)
+		return
+	}
+
+	type cmdDetails struct {
+		cmd     string
+		details []interface{}
+	}
+
+	cmdCount := len(reply) / 2
+	details := make([]cmdDetails, cmdCount)
```

Comment:
nit: better name for the variable would be `cmdDetails` IMO.

You might not even need the struct, just declare the details array as 

```
        cmdDetails := []struct {
		cmd     string
		details []interface{}
	}{}

Reply:
But bigger question: why do this in two steps and not just parse and submit the constHistogram instead of storing the intermediate result first?

Reply:
My original reasoning was to make it more readable, I now realize after having changed it that it didn't really make a difference on the front :)